test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Internal;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.AspNetCore.Sockets;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests
+{
+    public class HubEndpointTests
+    {
+        [Fact]
+        public async Task HubsAreDisposed()
+        {
+            var serviceProvider = CreateServiceProvider();
+            var endPoint = serviceProvider.GetService<HubEndPoint<TestHub>>();
+
+            using (var connectionWrapper = new ConnectionWrapper())
+            {
+                var endPointTask = Task.Run(() => endPoint.OnConnectedAsync(connectionWrapper.Connection));
+
+                Thread.Sleep(100);
+
+                // kill the connection
+                connectionWrapper.Connection.Channel.Dispose();
+
+                await endPointTask;
+
+                Assert.Equal(2, TestHub.DisposedCount);
+            }
+        }
+
+        private class TestHub : Hub
+        {
+            static public int DisposedCount = 0;
```

Comment:
I think you should pass in an object that has the state and not use a static for this.